CITATION.cff

```diff
@@ -0,0 +1,186 @@
+cff-version: 1.1.0
+title: "libCEED: the CEED Library"
```

Comment:
Do we want this to be our tagline? I know it's in the README, but it seems circular and uninformative. Just throwing out ideas:

1. libCEED: CEED library for Efficient Extensible Discretization
2. libCEED: Community library for Efficient Extensible Discretization
3. libCEED: Code for Efficient Extensible Discretization  *(the original)*
4. libCEED: Efficient Extensible Discretization

Reply:
Is the a `CITATION.cff` convention for escaping? The bibtex has
```
title = {libCEED: the CEED Library},
```
which will not capitalize correctly.

Reply:
I think it will put in whatever we have it the quotes literally.

I agree that this tagline isn't great. We could replace both at the same time. I like 3 or 4 personally but perhaps we should poll others?

Reply:
Update: escaping in the string has some mildly silly results:
```
@misc{Thompson_{LFAT}oolkit.jl_2021,
author = {Thompson, Jeremy L},
doi = {10.5281/zenodo.5064056},
month = {7},
title = {{LFAT}oolkit.jl},
url = {https://jeremylt.github.io/LFAToolkit.jl/stable/},
year = {2021}
}
```

Reply:
https://github.com/citation-file-format/citation-file-format/issues/258

Reply:
I'll resolve this. You can follow the above issue for proper case handling, which looks like it may take a while.